### ISSUE TITLE
Accessibility improvements

### ISSUE THREAD

**Issue by juuso-vaadin** [vaadin-component-factory/vcf-anchor-nav#7]
Each `vcf-anchor-nav-section` should have `role="region"` set.

In addition there should be `aria-labelledby` on the section and `id` in the header element.
Unfortunately linking the default header cannot be done with the current structure as the header element is within shadowDOM.

Consider moving the header element to light DOM to allow use of `aria-labelledby`.
